pymc/model/core.py

```diff
@@ -1611,6 +1611,7 @@ def compile_fn(
         inputs: Sequence[Variable] | None = None,
         mode=None,
         point_fn: bool = True,
+        borrow_vars: bool = False,
```

Comment:
Why an option and with this default? 

It shouldn't need to be an option. The point is that if input=output there's no need for a deepcopy. Step samplers always copy the point before mutating it, it's not supposed to be mutated inplace

Also, no reason to treat any differently than NDarray backend

Reply:
Ah this is happening inside `Model`. I didn't see the file header...

Reply:
We don't need these changes, as the `BaseTrace` class is the one compiling the function and it doesn't go through here. So no code duplication. Was there one in a previous version when I left my comment?

I would revert these changes in model core.py. They are not being used. The BaseTrace sidesteps Model altogether to create the point function

Reply:
Is this wrong then? I left it as `False` by default to keep backwards compatibility.

Reply:
You seem to have reintroduced the changes in `model/core.py`, I thought you had agreed to reverse them?